### PR TITLE
[e2e] Support IPv6 single stack clusters

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
[e2e] Support IPv6 single stack clusters
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/service_annotations.go [hunk 1]
```diff
@@ -232,20 +195,48 @@ var _ = Describe("Service with annotation", Label(utils.TestSuiteLabelServiceAnn
...
+		var newSubnetCIDR *net.IPNet
 		for _, existingSubnet := range *vNet.Subnets {
-			if *existingSubnet.Name == subnetName {
-				By("Test subnet have existed, skip creating")
-				newSubnetCIDR = *existingSubnet.AddressPrefix
-				break
+			if *existingSubnet.Name != subnetName {
+				continue
 			}
+			By("Test subnet have existed, skip creating")
```
Comment (nilo19) [inline tests/e2e/network/service_annotations.go:203]: Can we use utils.Logf here?

Comment (nilo19) [reply tests/e2e/network/service_annotations.go:203]: Why do we change the logic here?

Comment (lzhecheng) [reply tests/e2e/network/service_annotations.go]: Sure

**Pull request merged**